project/ScalastyleInspectionsGenerator.scala

```diff
@@ -0,0 +1,125 @@
+ */
+object ScalastyleInspectionsGenerator {
+
+  val generatorTask = Def.task {
+    val log = streams.value.log
+    log.info("Generating Scalastyle inspections file.")
+
+    val c = classOf[MainConfig]
+    val definitionStream: InputStream = c.getResourceAsStream("/scalastyle_definition.xml")
+    val documentationStream: InputStream = c.getResourceAsStream("/scalastyle_documentation.xml")
+    val inspections: NodeSeq = XML.load(definitionStream) \\ "checker"
+    val docs: NodeSeq = XML.load(documentationStream) \\ "scalastyle-documentation" \ "check"
+    val config: Config = ConfigFactory.load(c.getClassLoader)
+
+    // Collect Scalastyle inspections from config files.
+    val generatedInspections: Seq[ScalastyleInspection] = extractInspections(inspections, docs, config)
+
+    // Load the template file from ./project/ScalastyleInspections.scala.
+    val projectDir = new File(baseDirectory.value, "project")
+    val templateFile = new File(projectDir, "ScalastyleInspections.scala")
+
+    // Substitute AllInspections with generated inspections.
+    val source: Source = templateFile.parse[Source].get
+    val transformed = transform(source, generatedInspections)
+
+    // Save the new file to the managed sources dir.
+    val scalastyleInspectionsFile = (sourceManaged in Compile).value / "scalastyle" / "inspections.scala"
+    IO.write(scalastyleInspectionsFile, transformed.syntax)
+
+    Seq(scalastyleInspectionsFile)
+  }
+
+  /**
+   * Extract Scalastyle inspections from config files.
+   */
+  private[scalastyle] def extractInspections(
+    inspections: NodeSeq,
+    docs: NodeSeq,
+    config: Config
+  ): Seq[ScalastyleInspection] = {
+    val docsMap: Map[String, Node] = docs.map(node => node \@ "id" -> node).toMap
+    for {
+      inspection <- inspections
+      clazz = (inspection \ "@class").text
+      id = inspection \@ "id"
+      cfg = config.getConfig(id)
+      label = cfg.getString("label")
+      description = cfg.getString("description")
+      doc = docsMap.get(id)
+      extraDescription = doc.flatMap(node => (node \ "extra-description").map(_.text.trim).headOption)
+      justification = doc.map(node => (node \ "justification").text.trim)
+      defaultLevel = Level(inspection \@ "defaultLevel")
+      // TODO: Add template parameters.
+    } yield ScalastyleInspection(clazz, id, label, description, extraDescription, justification, defaultLevel)
+  }
+
+  /**
+   * Fill the template with generated inspections.
+   */
+  private[scalastyle] def transform(source: Tree, inspections: Seq[ScalastyleInspection]): Tree = {
+    val stringified: Seq[String] = inspections.collect {
+      case inspection =>
+        // TODO: Do we want to format the text with code examples?
+        // TODO: Is there a better way of embedding multi-line text?
+        val extraDescription = inspection.extraDescription.map(s => "\"\"\"" + s + "\"\"\"")
```

Comment:
Maybe ` s""""$s""""` is just sufficient?

Reply:
Thanks for all of your suggestions - I guess another alternative would be to convert new lines into `<br>`s and that way we could just wrap the whole text in single quotes, but I'm not going to worry about it for now. As I mentioned in the comment above, I'll come back to this in a separate PR when I can see how the raw formatting looks in the UI.

Reply:
ok